proto/backend.proto

```diff
@@ -0,0 +1,217 @@
+syntax = "proto3";
+package grafana.plugin;
+
+//-----------------------------------------------
+// Common
+//-----------------------------------------------
+
+message PluginConfig {
+  int64 id = 1;
+  int64 orgId = 2;
+  string name = 3;
+  string type = 4;
+  string url = 5;
+  string jsonData = 6;
+  map<string,string> decryptedSecureJsonData = 7;
+}
+
+//-----------------------------------------------
+// Plugin Status Check
+//-----------------------------------------------
+
+message PluginStatusRequest {
+  // Plugin Configuration
+  PluginConfig config = 1;
+
+  // Environment info
+  map<string,string> headers = 2;
+
+  enum FetchInfo {
+    STATUS = 0;  // nothing
+    API = 1;     // OpenAPI description
+    METRICS = 2; // Prometheus style metrics
+    DEBUG = 3;   // JSON Debug Info (admin+ view)
+  }
+
+  // Get an OpenAPI description (viewer+)
+  FetchInfo fetch = 3;
+}
+
+message PluginStatusResponse {
+  enum PluginStatus {
+    UNKNOWN = 0; // starting etc
+    OK = 1;
+    ERROR = 2;
+  }
+
+  PluginStatus status = 1;
+  string info = 2;
+}
+
+
+//-----------------------------------------------
+// QUERY
+//-----------------------------------------------
+
+message TimeRange {
+  string fromRaw = 1;
+  string toRaw = 2;
+  int64 fromEpochMs = 3;
+  int64 toEpochMs = 4;
+}
+
+message DataQuery {
+  string refId = 1;
+  int64 maxDataPoints = 2;
+  int64 intervalMs = 3;
+  TimeRange timeRange = 4; // NOTE: this moved from QueryRequest in v1
+  string json = 5; // Arbitrary structure (model?  modelJson in v1)
+}
+
+message QueryRequest {
+  // Plugin Configuration
+  PluginConfig config = 1;
+
+  // Environment info
+  map<string,string> headers = 2;
```

Comment:
In most instances when we have headers in a query and response I think we should consider having two fields. grafanaHeaders and customHeaders. This way we don't have to worry about name collision for "reserved headers". I think it also might make for better patterns in the SDK.